src/schema/notifications.ts

```diff
@@ -49,6 +67,68 @@ export class ZimbraNotifications {
 		this.handleFolderNotifications(notification);
 		this.handleConversationNotifications(notification);
 		this.handleMessageNotifications(notification);
+		this.handleContactNotifications(notification);
+	};
+
+	private handleContactNotifications = (notification: Notification) => {
+		const items = itemsForKey(notification, 'cn');
+		if (items) {
+			const searchContactResponse: any = this.cache.readFragment({
+				id: `${findDataId(this.cache, '$ROOT_QUERY.search', dataId =>
+					/in:\\"Contacts\\" NOT #type:group/.test(dataId)
+				)}`,
+				fragment: gql`
+					fragment ${generateFragmentName('searchResults')} on SearchResponse {
+						contacts
+					}
+				`
+			});
+			let newContacts: any[] = [];
+			items.forEach((i: any) => {
+				const item = normalizeContact(i);
+				this.cache.writeFragment({
+					id: `Contact:${item.id}`,
+					fragment: gql`
+						fragment ${generateFragmentName('contactNotification', item.id)} on Contact {
+							${attributeKeys(item)}
+						}
+					`,
+					data: {
+						__typename: 'Contact',
+						...item
+					}
+				});
+				if (
+					searchContactResponse.contacts &&
+					!searchContactResponse.contacts.some(
+						(contact: any) => contact.id === `Contact:${item.id}`
+					)
+				) {
+					newContacts = [
+						{
+							generated: false,
+							id: `Contact:${item.id}`,
+							type: 'id',
+							typename: 'Contact'
+						}
+					].concat(newContacts);
+				}
+			});
+			this.cache.writeFragment({
+				id: `${findDataId(this.cache, '$ROOT_QUERY.search', dataId =>
+					/in:\\"Contacts\\" NOT #type:group/.test(dataId)
```

Comment:
I talked with @aschneit about this on hipchat: we'll have to lookup the folder name that the contact was created in and then write it to that folder.

Right now creating a new contact in an alternate folder (like "Emailed Contacts") causes the contact to appear in the main "Contacts" folder.